developer-docs/README.md

```diff
@@ -64,102 +64,207 @@ SEMATIC_API_ADDRESS: https://<my_prod1_host:port>
+performed in order, without skipping any. If any errors occur, you must fix
+them and then go back and redo all the steps that have been affected by the
+changes.
 
-```bash
-HELM_REPO=~/code/helm-charts
-helm package helm/sematic-server
-helm repo index . --url https://sematic-ai.github.io/helm-charts/sematic-server \
-         --merge $HELM_REPO/index.yaml
-mv index.yaml $HELM_REPO/index.yaml
-mv *.tgz $HELM_REPO/sematic-server/
-```
+1. Bump the version in:
+   - `wheel_constants.bzl` - `wheel_version_string`
+   - `helm/sematic-server/Chart.yaml` - `appVersion`
+   - `sematic/versions.py` - `CURRENT_VERSION`; bump `MIN_CLIENT_SERVER_SUPPORTS` if
+     there are any TODOs mentioning this should be done
+   - `./README.md` - PyPI badge
 
-You should now have a new `sematic-server/sematic-server-X.X.X.tgz` in the
-`helm-charts` repo, and a modified `index.yaml`.  Commit and push both of
-these to a new branch, and create a PR for the change based on `gh-pages`
-necessary.
-
-Finally, draft the release on GitHub:
-- Add a "What's Changed" section.
-- Add a warning that the client version will need to be upgraded, if this applies.
-- Add a "New Contributors" section, if this applies.
-- Add a "Full Changelog" link.
-- Attach the wheel in the "Assets" section.
+1. Increment the minor version of the `version` field in
+  `helm/sematic-server/Chart.yaml`. Note that this will be DIFFERENT
+  from the Sematic version you changed in prior steps (chart version
+  will be `1.X.Y`, Sematic version is `0.M.N`).
+
+1. Update `changelog.md` with the new version number and any missing change entries.
+
+1. Make the release PR, containing the previous changes. After implementing
+  comments and getting approval on the release PR, merge it and pull from the
+  updated `main`. **It is mandatory to include this version of main in the
+  subsequent steps. If validation issues are discovered, they must be fixed
+  in patch PRs, and the process restarted from this step.**
+
+1. Build the UI:
+    ```bash
+    $ make ui
+    ```
+
+1. Build the wheel:
+    ```bash
+    $ make wheel
+    ```
+
+1. Test the Server wheel locally for all supported versions of Python.
+
+    1. Copy the wheel from `bazel-bin/sematic/sematic-*.whl` into a scratch
+    directory.
+
+    1. For each supported version of Python, use a virtual env to test:
+      ```bash
+      $ # LOCALLY:
+      $ pip3 install <wheel path>
+      $ sematic stop
+      $ sematic version # check that the correct sematic and python versions are used
+      $ sematic start
+      $ sematic run examples/mnist/pytorch
+      $ sematic stop
+      ```
+
+1. Test the [internal Helm charts](/helm/sematic-server) deployment.
+
+    1. **Test an upgrade deployment** in the `stage` environment, **where the previous
+    version is already deployed**, use
+    [`serve-dev`](https://github.com/sematic-ai/infrastructure/tree/main/bin) to upgrade
```

Comment:
Should we mention that the following steps need to be done in a different repository? (The above steps all belong to this repo).

Reply:
The sentence mentions this must be done in the stage environment, and the first line of the commands snippet below is a comment with the stage environment.

Reply:
I am not sure it is the same thing. Does `stage` environment imply that the following commands are accessed from the `infrastructure` repository? 

If we expect readers to figure it out from the link of `serve-dev`, it might not be that obvious.

Reply:
These commands don't need to be ran from the infrastructure repository; they can be ran from anywhere.

Reply:
Yes, indeed. However, do we expect the person who handles the release to download the raw file from the infrastructure repo? Or do we expect him to git clone the repo? Or do we expect him to figure out how to run the script all on his own? 

These questions might make him disoriented. It might help to clarify a little bit.

Reply:
The link on this line points to the repo which contains a README file that explains why this is necessary and how to use it. Even if a new dev is trying to perform the release without having had read the documentation linked in these steps (cause enough for tar and feathering), when they would try to execute the `serve-dev` command, they would get a not found error, forcing them to read the documentation.